### PR TITLE
fix / skip (for now) some tests before switch to torch 2.2

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Some changes are necessary to switch to 2.2. Others are flaky even in torch 2.1.

### CODE REVIEW

File: tests/models/mega/test_modeling_mega.py [hunk 1]
```diff
@@ -534,6 +535,16 @@ def setUp(self):
...
 
+    # TODO: @ydshieh
+    @is_flaky(description="Sometimes gives `AssertionError` on expected outputs")
+    def test_pipeline_fill_mask(self):
+        super().test_pipeline_fill_mask()
+
+    # TODO: @ydshieh
+    @is_flaky(description="Sometimes gives `RuntimeError: probability tensor contains either `inf`, `nan` or element < 0`")
+    def test_pipeline_text_generation(self):
+        super().test_pipeline_text_generation()
```
Comment (ydshieh) [inline tests/models/mega/test_modeling_mega.py:546]: flaky even with torch 2.1.2 (IIRC)

File: tests/models/vits/test_modeling_vits.py [hunk 2]
```diff
@@ -176,6 +176,11 @@ def setUp(self):
     def test_config(self):
         self.config_tester.run_common_tests()
 
+    # TODO: @ydshieh
+    @is_flaky(description="torch 2.2.0 gives `Timeout >120.0s`")
+    def test_pipeline_feature_extraction(self):
+        super().test_pipeline_feature_extraction()
```
Comment (ydshieh) [inline tests/models/vits/test_modeling_vits.py:182]: torch 2.2 issue

File: tests/utils/test_model_output.py [hunk 3]
```diff
@@ -155,9 +155,11 @@ def test_torch_pytree(self):
         if is_torch_greater_or_equal_than_2_2:
             self.assertEqual(
                 pytree.treespec_dumps(actual_tree_spec),
-                '[1, {"type": "tests.utils.test_model_output.ModelOutputTest", "context": ["a", "c"], "children_spec": [{"type": null, "context": null, "children_spec": []}, {"type": null, "context": null, "children_spec": []}]}]',
+                '[1, {"type": "tests.utils.test_model_output.ModelOutputTest", "context": "[\\"a\\", \\"c\\"]", "children_spec": [{"type": null, "context": null, "children_spec": []}, {"type": null, "context": null, "children_spec": []}]}]',
```
Comment (ydshieh) [inline tests/utils/test_model_output.py:158]: I forgot to check this test with torch 2.2 and it was never run before.
Here is the correct output.

File: tests/utils/test_model_output.py [hunk 4]
```diff
@@ -155,9 +155,11 @@ def test_torch_pytree(self):
...
             )
 
+    # TODO: @ydshieh
+    @unittest.skip("CPU OOM")
```
Comment (ydshieh) [inline tests/utils/test_model_output.py:162]: will check what happen later. Not very urgent for now

**Comment by ydshieh** [comment]
Note, this PR doesn't unpin torch

**Review by amyeroberts — APPROVED** [review]
Thanks for adding all these guards and helping our CI run smoothly 🙏

**Pull request merged**